kadai4/yuonoda/lib/omikuji.go

```diff
@@ -0,0 +1,50 @@
+package omikuji
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+var results = []string{"凶", "吉", "中吉", "大吉"}
+
+type Omikuji struct {
+	Result string `json:"result"`
+}
+
+func pickResult() string {
+	_, m, d := time.Now().Date()
+	var i int
+	if m == 1 && d > 0 && d < 4 {
+		i = len(results) - 1
+	} else {
+		i = rand.Intn(len(results))
+	}
+	return results[i]
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	result := pickResult()
+	o := &Omikuji{Result: result}
+
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	if err := enc.Encode(o); err != nil {
+		log.Fatal(err)
```

Comment:
課題がAPIの実装なので、API側で処理に失敗した場合はレスポンスで失敗したことを伝えたほうがいいかもしれません。

レスポンスの一例を載せておきます。ご参考になれば幸いです。
```suggestion
                log.Print(err)
		http.Error(w, "処理中にエラーが発生しました。もう一度やり直してください", http.StatusInternalServerError)
```